modules/tracing/mule-tracer-customization-api/src/main/java/org/mule/runtime/tracer/configuration/api/InitialSpanInfoProvider.java

```diff
@@ -38,6 +38,14 @@ public interface InitialSpanInfoProvider {
   InitialSpanInfo getInitialSpanInfo(String name);
 
 
+  /**
+   * @param name the name for the span.
+   *
+   * @return a {@link InitialSpanInfo} based on a name taking into account that this is a debug level span.
+   */
+  InitialSpanInfo getDebugLevelInitialSpanInfo(String name);
```

Comment:
I think we made a previous mistake by adding the method getInitialSpanInfo(name) here. There might be one or two corner cases but It's not that we do not have a component, we are just not passing it. That error is becoming evident now that we need to map levels and overrides to a set of components that will be traced and exported. Can we consider switching to such API and try to leave the level logic only at the export layer? I think that it will be very convenient for the next steps. On the other hand, we can also add a task to do it, but I hate technical debt and it looks that we will need to tackle it soon anyways.